modules/tinymce/src/plugins/link/main/ts/ui/sections/LinkListOptions.ts

```diff
@@ -27,18 +26,22 @@ const getLinks = (editor: Editor): Promise<Optional<ListItem[]>> => {
   const extractor = (item: UserListItem) => editor.convertURL(item.value || item.url, 'href');
 
   const linkList = Options.getLinkList(editor);
-  return new Promise<Optional<UserListItem[]>>((callback) => {
+  return new Promise<Optional<UserListItem[]>>((resolve) => {
     // TODO - better handling of failure
     if (Type.isString(linkList)) {
-      XHR.send({
-        url: linkList,
-        success: (text) => callback(parseJson(text)),
-        error: (_) => callback(Optional.none())
-      });
+      window.fetch(linkList)
+        .then((res) => {
+          if (res.ok) {
+            res.text().then((text) => resolve(parseJson(text)));
+          } else {
+            resolve(Optional.none());
+          }
+        })
+        .catch(() => resolve(Optional.none()));
```

Comment:
This doesn't handle the case where `res.text()` throws an error. I think we'd be better doing something like this maybe?
```suggestion
        .then((res) => res.ok ? res.text().then(parseJson) : Promise.reject())
        .then(resolve, () => resolve(Optional.none()));
```

Reply:
This is better. Thanks